src/lazy/encoder/writer.rs

```diff
@@ -100,7 +100,10 @@ impl<E: Encoding, Output: Write> Writer<E, Output> {
     /// Writes bytes of previously encoded values to the output stream.
     pub fn flush(&mut self) -> IonResult<()> {
         if self.context.num_pending_symbols > 0 {
-            self.write_lst_append()?;
+            match E::ion_version() {
+                IonVersion::v1_0 => self.write_lst_append()?,
+                IonVersion::v1_1 => self.write_encoding_directive()?,
+            }
```

Comment:
🗺️ The `Writer` now writes either an LST or encoding directive depending on the Ion version of the stream.